### PR TITLE
feat(dev-liquidity): moves from uniswap to coingecko

### PR DESCRIPTION
**Pull request by jurajpiar**
Replaces uniswap for coingecko faking server endpoints.

Ref: https://rsklabs.atlassian.net/browse/ROLLUP-147

### CODE REVIEW

**Review by patogallaiovlabs — DISMISSED** [review]

File: core/bin/zksync_api/src/fee_ticker/mod.rs
Comment (jurajpiar) [reply core/bin/zksync_api/src/fee_ticker/mod.rs:55]: it is. the mod file exports it so that it can be used elsewhere (such as [core/bin/zksync_api/src/bin/dev-liquidity-token-watcher.rs#L15](https://github.com/rsksmart/ri-aggregation/blob/feature/dev_coingecko_watcher/core/bin/zksync_api/src/bin/dev-liquidity-token-watcher.rs#L15))

File: core/bin/zksync_api/src/bin/dev-liquidity-token-watcher.rs
Comment (jurajpiar) [reply core/bin/zksync_api/src/bin/dev-liquidity-token-watcher.rs]: done

File: core/bin/zksync_api/src/bin/dev-liquidity-token-watcher.rs
Comment (antomor) [inline core/bin/zksync_api/src/bin/dev-liquidity-token-watcher.rs:131]: Why do we create many services instead of creating one single service and many routes? What's the reasoning behind it? I'd consider using many routes within the same service,

File: core/bin/zksync_api/src/bin/dev-liquidity-token-watcher.rs
Comment (antomor) [inline core/bin/zksync_api/src/bin/dev-liquidity-token-watcher.rs:171]: I like this

File: core/bin/zksync_api/src/bin/dev-liquidity-token-watcher.rs
Comment (antomor) [inline core/bin/zksync_api/src/bin/dev-liquidity-token-watcher.rs:247]:
```suggestion
        .unwrap(); // is this really the simplest way to get the body? seems so unnecesary to convert it to BoxAnyBody, etc!
```

File: core/bin/zksync_api/src/bin/dev-liquidity-token-watcher.rs
Comment (antomor) [inline core/bin/zksync_api/src/bin/dev-liquidity-token-watcher.rs:293]:
```suggestion
        .unwrap(); // is this really the simplest way to get the body? seems so unnecesary to convert it to BoxAnyBody, etc!
```

File: core/bin/zksync_api/src/bin/dev-liquidity-token-watcher.rs
Comment (antomor) [inline core/bin/zksync_api/src/bin/dev-liquidity-token-watcher.rs:250]:
```suggestion
            let platform_option = body
```

File: docker-compose.yml
Comment (antomor) [inline docker-compose.yml:27]: I'd remove it if not necessary, or we add a more meaningful comment, like `TODO: Once we publish the image, we could use it directly, instead of building it each time.

File: docker/dev-liquidity-token-watcher/Dockerfile
Comment (antomor) [inline docker/dev-liquidity-token-watcher/Dockerfile:6]:
```suggestion
COPY . .
```

File: infrastructure/zk/src/docker.ts
Comment (antomor) [inline infrastructure/zk/src/docker.ts:48]: Why is this necessary? Why do we want to tag it differently? I'd change it once we manage to publish the images under `rsksmart`

File: infrastructure/zk/src/docker.ts
Comment (antomor) [inline infrastructure/zk/src/docker.ts:78]: Could we please add a comment here to specify that we need to restore this change once we publish the image?

File: core/bin/zksync_api/src/bin/dev-liquidity-token-watcher.rs
Comment (antomor) [inline core/bin/zksync_api/src/bin/dev-liquidity-token-watcher.rs:144]: I think we need to review the configuration to remove the `graphql` suffix

**Review by antomor — CHANGES_REQUESTED** [review]
@jurajpiar thank you for the changes. I haven't tried it locally, but the changes look fine. I plan to run it locally before merging it. For now I've just left some minor comments/doubts.

**Review by antomor — CHANGES_REQUESTED** [review]
@jurajpiar thank you for the changes. I haven't tried it locally, but the changes look fine. I plan to run it locally before merging it. For now I've just left some minor comments/doubts.

File: etc/env/docker.env
Comment (jurajpiar) [reply etc/env/docker.env:11]: yea I ran this without an issue afaik. besides this is simply copied from the orig config. why shouldn't it work? The host/guest ports are bound together aren't they?

Comment (antomor) [reply etc/env/docker.env:11]: Yes, I think it works if you run the watcher directly, while it won't work if you run the watcher within docker. if you say that it worked, it means that `ETH_CLIENT_WEB3_URL` isn't used from the watcher. Could we confirm?

File: core/bin/zksync_api/src/bin/dev-liquidity-token-watcher.rs
Comment (franciscotobar) [inline core/bin/zksync_api/src/bin/dev-liquidity-token-watcher.rs:99]:
```suggestion
    if !keys.is_empty() {
```

Comment (jurajpiar) [reply core/bin/zksync_api/src/bin/dev-liquidity-token-watcher.rs:99]: done thanks

File: infrastructure/zk/src/docker.ts
Comment (jurajpiar) [reply infrastructure/zk/src/docker.ts:78]: done

File: infrastructure/zk/src/docker.ts
Comment (jurajpiar) [reply infrastructure/zk/src/docker.ts:48]: removed

File: docker/dev-liquidity-token-watcher/Dockerfile
Comment (jurajpiar) [reply docker/dev-liquidity-token-watcher/Dockerfile:6]: done

File: core/bin/zksync_api/src/bin/dev-liquidity-token-watcher.rs
Comment (jurajpiar) [reply core/bin/zksync_api/src/bin/dev-liquidity-token-watcher.rs:250]: done

File: core/bin/zksync_api/src/bin/dev-liquidity-token-watcher.rs
Comment (jurajpiar) [reply core/bin/zksync_api/src/bin/dev-liquidity-token-watcher.rs:247]: done

File: core/bin/zksync_api/src/bin/dev-liquidity-token-watcher.rs
Comment (jurajpiar) [reply core/bin/zksync_api/src/bin/dev-liquidity-token-watcher.rs:293]: done

File: etc/env/docker.env
Comment (jurajpiar) [reply etc/env/docker.env]: done

File: etc/env/docker.env
Comment (antomor) [inline etc/env/docker.env:10]: This value is present twice, see line #1

File: core/bin/zksync_api/src/bin/dev-liquidity-token-watcher.rs
Comment (antomor) [reply core/bin/zksync_api/src/bin/dev-liquidity-token-watcher.rs:131]: My fault, I didn't realize before that the services are nested.

File: core/bin/zksync_api/src/bin/dev-liquidity-token-watcher.rs
Comment (antomor) [inline core/bin/zksync_api/src/bin/dev-liquidity-token-watcher.rs:247]:
```suggestion
        .unwrap(); // is this really the simplest way to get the body? seems so unnecessary to convert it to BoxAnyBody, etc!
```

File: core/bin/zksync_api/src/bin/dev-liquidity-token-watcher.rs
Comment (antomor) [inline core/bin/zksync_api/src/bin/dev-liquidity-token-watcher.rs:293]:
```suggestion
        .unwrap(); // is this really the simplest way to get the body? seems so unnecessary to convert it to BoxAnyBody, etc!
```

**Review by antomor — APPROVED** [review]
LGTM

**Comment by jurajpiar** [comment]
removed the "do not merge" label from this PR as it merges into another feature branch.

File: etc/env/docker.env
Comment (jurajpiar) [reply etc/env/docker.env:9]: done

File: core/bin/zksync_api/src/bin/dev-liquidity-token-watcher.rs
Comment (jurajpiar) [reply core/bin/zksync_api/src/bin/dev-liquidity-token-watcher.rs]: done

File: core/bin/zksync_api/src/bin/dev-liquidity-token-watcher.rs
Comment (jurajpiar) [reply core/bin/zksync_api/src/bin/dev-liquidity-token-watcher.rs]: done

**Pull request merged**
